src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,344 @@
+     *
+     * @return The collection
+     */
+    public ArrayList<Task> getTasks() {
+        return this.tasks;
+    }
+    /**
+     * displays the list of Tasks
+     */
+    public void displayList() {
+        System.out.println(indent + "Here are the tasks in your list:");
+        for (int i = 0; i < tasks.size(); i++) {
+            int num = i + 1;
+            Task curr = tasks.get(i);
+            System.out.println(indent + num + "." + curr.toString());
+        }
+    }
+    /**
+     * Displays the list of Tasks that description matches the user input
+     * @param userInput the String that the user inputs to find similar Tasks
+     */
+    public void displayMatchingList(String userInput) {
+        ArrayList<Task> temp = new ArrayList<>();
+        for (int i = 0; i < tasks.size(); i++) {
+            Task task = tasks.get(i);
+            String taskDescription = task.getDescription();
+            if (taskDescription.contains(userInput)) {
+                temp.add(task);
+            }
+        }
+        if (temp.size() == 0) {
+            System.out.println(indent + "There are no matching tasks");
+        } else {
+            System.out.println(indent + "Here are the matching tasks in your list:");
+            for (int i = 0; i < temp.size(); i++) {
+                int num = i + 1;
+                Task curr = temp.get(i);
+                System.out.println(indent + num + "." + curr.toString());
+            }
+        }
+    }
+    /**
+     * This method encapsulates the functionality of marking a task as completed or not
+     * For example, the input 'mark 1' will mark the duke.Task at position 0 at the TaskArray as 'marked'
+     * @param string the input string
+     * @throws DukeException if input is invalid
+     */
+    public void markDescription(String string) throws DukeException {
+        String clean = string.replaceAll("\\D+", ""); //remove non-digits
+        int pos = Integer.parseInt(clean) - 1;
+        if (pos >= tasks.size()) {
+            throw new DukeException("You are trying to access a Task that does not exist!");
+        }
+        Task curr = tasks.get(pos);
+
+        if (string.contains("unmark")) {
+            curr.markAsUnDone();
+            System.out.println(indent + "OK, I've marked this task as not done yet:");
+        } else if (string.contains("mark")) {
+            curr.markAsDone();
+            System.out.println(indent + "Nice! I've marked this task as done:");
+        }
+        System.out.println(megaIndent + curr.getStatusIconWithBracket() + " " + curr.description);
+    }
+    /**
+     * For deadline and event Tasks, obtains the description of the duke.Task (before the first slash)
+     * For example, the input 'event project meeting /from Mon 2pm /to 4pm' will return 'project meeting'
+     *
+     * @param string of the duke.Task
+     * @return the description of the duke.Task
+     */
+    public static String getDescription(String string) {
```

Comment:
Likewise here as well, maybe there can be a better parameter naming instead of "string"